OpenRA.Mods.Common/Activities/Move/MoveWithinRange.cs

```diff
@@ -53,5 +54,18 @@ bool AtCorrectRange(WPos origin)
 		{
 			return Target.IsInRange(origin, maxRange) && !Target.IsInRange(origin, minRange);
 		}
+
+		bool InCorrectLayer(Actor self)
```

Comment:
The duplication of this method would suggest `Target.IsInCorrectLayer` could be a good method to have.